### PR TITLE
Observe config v2 for all documented node configs

### LINKED ISSUE #206: Upgrade to dbt_project config-version: 2 syntax
This is going to be a _ton_ of tiny changes. E.g. one that tripped me up today:

[This](https://docs.getdbt.com/reference/resource-configs/column_types#usage)
```yml
seeds:
  jaffle_shop:
    country_codes:
      column_types:
        country_code: varchar(2)
        country_name: varchar(32)
```
needs to become
```yml
seeds:
  jaffle_shop:
    country_codes:
      +column_types:
        country_code: varchar(2)
        country_name: varchar(32)
```

### PR DESCRIPTION
**Pull request by jtcohen6**
* resolves #206
* Update all docs around node configs in `dbt_project.yml` to observe `config-version: 2`, introduced in 0.17.0
* Understood if we want to wait on this. My assumption is that, going forward, all _brand new_ dbt users will be on >= v0.17.0

### CODE REVIEW

**Comment by drewbanin** [comment]
https://github.com/fishtown-analytics/docs.getdbt.com/pull/221/files
https://github.com/fishtown-analytics/docs.getdbt.com/pull/219

👀 👀 

Will let you two figure out how to proceed here :)

File: website/docs/docs/building-a-dbt-project/building-models.md [hunk 1]
```diff
@@ -115,11 +115,11 @@ name: jaffle_shop
```
Comment (clrcrl) [inline website/docs/docs/building-a-dbt-project/building-models.md:115]:
```suggestion
config-version: 2

```

File: website/docs/learn/3-build-your-first-models.md [hunk 2]
```diff
@@ -71,9 +71,9 @@ One of the most powerful features of dbt is that you can change the way a model
 ```yml
 models:
   jaffle_shop:
-    materialized: table
+    +materialized: table
```
Comment (clrcrl) [inline website/docs/learn/3-build-your-first-models.md:74]: The one thing i'm not sure about is including it in the tutorial / learn onboarding — the [starter-project](https://github.com/fishtown-analytics/dbt-starter-project/blob/master/dbt_project.yml) generates a v1 syntax, and the videos are in v1 syntax. Not sure what to do here!

Comment (jtcohen6) [reply website/docs/learn/3-build-your-first-models.md:74]: Good point. Do we have a timeline on upgrading the starter project? I see that Jake added a [tag](https://github.com/fishtown-analytics/dbt-starter-project/releases/tag/dbt-yml-config-version-2) for config v2 projects, I'm guessing that's what `dbt init` on >=0.17.0 clones instead

Comment (clrcrl) [reply website/docs/learn/3-build-your-first-models.md:74]: Just opened a [PR](https://github.com/fishtown-analytics/dbt-starter-project/pull/8) to fix the config on that repo 🙂 . I didn't realize we had updated the starter-script, I guess maybe it's a case of adding a little callout like "this differs from the video"?

Comment (clrcrl) [reply website/docs/learn/3-build-your-first-models.md:81]: I ended up adding some info — once that renders correctly, let's merge this in :) 
(I have some other PRs I want to merge in tomorrow that are going to conflict big time!)

**Comment by drewbanin** [comment]
I'm removing myself from review - we should ship this whenever you two feel good about it! Let me know if there's anything specific in here that you'd like another pair of eyes on thought

**Comment by jtcohen6** [comment]
@clrcrl Sorry again for interceding on your work on #219. Thanks for seeing this through!

**Pull request merged**